### ISSUE TITLE
Detector # issue

### ISSUE THREAD

**Issue by VinzenzBildstein** [GRIFFINCollaboration/NTuple2EventTree#8]
_From @scshadrick on November 20, 2017 21:37_

Updating from email discussions, the detector # is reported as -1 for all hits through grsiproof analysis.  The event tree in use is from detectorsims g4out.root -> NTuple2EventTree and run through grsiproof.

The DESCANT ID's have been 8010, 8020, 8030, 8040, and 8050.  

![4vinzenzoutputntuple2evtree](https://user-images.githubusercontent.com/20971746/33042738-1ce05366-ce00-11e7-9c84-3548c39dae22.png)

This shows the address not being set correctly.

![55](https://user-images.githubusercontent.com/20971746/33042745-26563b18-ce00-11e7-9db0-50811fb555fc.png)

This is the systemID + detNumber output from NTupletoEventTree in Converter.cc after the switch statement on (fSystemID).

_Copied from original issue: GRIFFINCollaboration/GRSISort#962_

**Comment by VinzenzBildstein** [comment]
Is that fSystemID and fDetNumber in the last screen shot? What about address, what has that been set to?

Also, if you open the resulting analysis*.root file and look at the events in there? 

Use something like:
```
TDescant* desc = nullptr;
AnalysisTree->SetBranchAddress("TDescant", &desc);
for(int i = 0; i < 10; ++i) { AnalysisTree->GetEntry(i); desc->Print(); }
```

Actually, looking at the code, I'm not sure how this worked for you at all. The analysis tree is currently being written as "EventTree", but all grsisort programs (including grsiproof) look for a tree called "AnalysisTree". So this should never have worked for you. How did you get the output of the first screen shot?

**Comment by scshadrick** [comment]
The output of the last screenshot is from execution of NTuple2EventTree with a cout statement in Converter::Run() after the switch statement.  The address is being set to 8000 + fDetNumber.  

I'm just using grsiproof for analysis by doing [$ grsiproof analysis*.root TreeSelector.C].  I still get the right multiplicity out of the Descant object from EventTree, so it recognizes it.  Where would we change GRSI to look for EventTree instead?

**Comment by scshadrick** [comment]
I looked into my code, I had just changed the last line in grsiproof.cxx to Analyze("EventTree") rather than AnaylsisTree.

**Comment by VinzenzBildstein** [comment]
I know the address **should** be set to 8000+fDetNumber, but it seems that grsiproof doesn't see that. So either the address didn't get set correctly, or there is some other issue.
Have you tried running the code I posted (with AnalysisTree changed to EventTree)?
And how did you get the first screenshot? Is that just a Print statement for every hit in the event?

BTW, I changed the program so that if writes the tree as AnalysisTree (see PR #9).

**Comment by scshadrick** [comment]
I figured it out.  In Converter.cc, *fFragment can be changed to just frag.second in each call of AddFragment.  The code just didn't like that *fFragment variable.

![66det](https://user-images.githubusercontent.com/20971746/33088302-8437523a-ceaa-11e7-9bc5-535b6937962f.png)

I now have this as output for det# from grsiproof analysis selector for all hits.  I read the verbosities of NTuple2EventTree and found that the address was set right for the fragments if I did the prescription I mentioned above.
